File: src/test/java/job/JobJobManagerTest.java
package job;

import org.junit.Test;

public class JobJobManagerTest {

    private static int POOL_SIZE = 2;

    @Test
    public void test() throws Exception {
        JobJobManager pjs = new JobJobManager(POOL_SIZE);//, QUEUE_SIZE);
        pjs.startManager();

        pjs.scheduleJob(new OwmsOutputCdrJob("Job1", Job.PRIORITY_LOW));
        Thread.sleep(100);
        pjs.scheduleJob(new OwmsOutputCdrJob("Job2", Job.PRIORITY_LOW));
        Thread.sleep(100);
        pjs.scheduleJob(new OwmsOutputCdrJob("Job3", Job.PRIORITY_LOW));
        Thread.sleep(100);
        pjs.scheduleJob(new OwmsOutputCdrJob("Job4", Job.PRIORITY_LOW));
        Thread.sleep(100);
        pjs.scheduleJob(new OwmsOutputCdrJob("Job5", Job.PRIORITY_LOW));
        Thread.sleep(100);
        pjs.scheduleJob(new OwmsOutputCdrJob("Job6", Job.PRIORITY_HIGH));

        pjs.getJobList().all().forEach(job -> System.out.println(job));


        pjs.cancelJob("Job5");

        //Thread.sleep(1000);
        pjs.scheduleJob(new OwmsOutputCdrJob("Job7", Job.PRIORITY_LOW));
        pjs.scheduleJob(new OwmsOutputCdrJob("Job8", Job.PRIORITY_HIGH));
        pjs.scheduleJob(new OwmsOutputCdrJob("Job9", Job.PRIORITY_LOW));
        pjs.scheduleJob(new OwmsOutputCdrJob("Job10", Job.PRIORITY_HIGH));

        while(true) {
            try { Thread.sleep(5000); }catch (Exception e) {e.printStackTrace();}
            System.out.println();
            pjs.getJobList().all().forEach(job -> System.out.println(job));
        }
    }
}

File: src/test/scala/JobJobManagerTest.scala
import org.scalatest.funsuite.AnyFunSuite

class JobJobManagerTest extends AnyFunSuite {

  test("startManager -> stopManager") {
    val m = new JobManager(2)
    m.startManager()
    Thread.sleep(10000)
    m.stopManager()
  }

  test("addJob") {
    val m = new JobManager(2)
    m.startManager()
    //Thread.sleep(1000)
    m.addJob(OwmsOutputCdrJob("job01", Job.PRIORITY_LOW, 0L, 100000L))
    m.addJob(OwmsOutputCdrJob("job02", Job.PRIORITY_HIGH, 3L, 100000L))
    m.addJob(OwmsOutputCdrJob("job03", Job.PRIORITY_HIGH, 4L, 100000L))
    Thread.sleep(5000)
    m.addJob(OwmsOutputCdrJob("job04", Job.PRIORITY_HIGH, 4L, 100000L))
    Thread.sleep(10000)

    m.jobStatusList.foreach(println)
    m.stopManager()
  }

  test("cancel") {
    val m = new JobManager(2)
    m.startManager()
    //Thread.sleep(1000)
    m.addJob(OwmsOutputCdrJob("job01", Job.PRIORITY_LOW, 0L, 100000L))
    m.addJob(OwmsOutputCdrJob("job02", Job.PRIORITY_HIGH, 3L, 100000L))
    m.addJob(OwmsOutputCdrJob("job03", Job.PRIORITY_HIGH, 4L, 100000L))

    Thread.sleep(100)
    m.cancelJob("job01")
    m.cancelJob("job02")
    Thread.sleep(3000)

    m.stopManager()
    m.jobStatusList.foreach(println)
  }

  test("timeout") {
    val m = new JobManager(2)
    m.startManager()
    //Thread.sleep(1000)
    m.addJob(OwmsOutputCdrJob("job01", Job.PRIORITY_LOW, 0L, 1000L))
    m.addJob(OwmsOutputCdrJob("job02", Job.PRIORITY_HIGH, 3L, 2000L))
    m.addJob(OwmsOutputCdrJob("job03", Job.PRIORITY_HIGH, 4L, 100000L))
    Thread.sleep(10000)

    m.jobStatusList.foreach(println)
    m.stopManager()
  }
}